x/crosschain/types/status.go

```diff
@@ -9,28 +9,47 @@ func (m *Status) AbortRefunded() {
 	m.StatusMessage = "CCTX aborted and Refunded"
 }
 
-// ChangeStatus changes the status of the cross chain transaction
-// empty msg does not overwrite old status message
-func (m *Status) ChangeStatus(newStatus CctxStatus, msg string) {
-	if len(msg) > 0 {
-		if m.StatusMessage != "" {
-			m.StatusMessage = fmt.Sprintf("%s : %s", m.StatusMessage, msg)
-		} else {
-			m.StatusMessage = msg
-		}
-	}
+// UpdateStatusAndErrorMessages transitions the Status and Error messages.
+func (m *Status) UpdateStatusAndErrorMessages(newStatus CctxStatus, isError bool, statusMsg, errorMsg string) {
+	m.UpdateStatus(newStatus, statusMsg)
+	m.UpdateErrorMessage(isError, errorMsg)
+}
+
+// UpdateStatus updates the cctx status and cctx.status.status_message.
+func (m *Status) UpdateStatus(newStatus CctxStatus, statusMsg string) {
 	if !m.ValidateTransition(newStatus) {
 		m.StatusMessage = fmt.Sprintf(
-			"Failed to transition : OldStatus %s , NewStatus %s , MSG : %s :",
+			"Failed to transition status from %s to %s",
 			m.Status.String(),
 			newStatus.String(),
-			msg,
 		)
+
 		m.Status = CctxStatus_Aborted
 		return
 	}
+
+	m.StatusMessage = fmt.Sprintf("Status changed from %s to %s", m.Status.String(), newStatus.String())
+
+	if statusMsg != "" {
+		m.StatusMessage += fmt.Sprintf(": %s", statusMsg)
```

Comment:
As previously answered, status_message usually carries the status transition or the error of transitioning it. For some exceptional cases it might carry some message specifying why the status had to be changed.
Error message should carry just the error so an external dev can get it easily.